### PR TITLE
BugFix: Azure now tracks of all VMs resources

### PR DESCRIPTION
**Pull request by prashanth26**
Azure earlier treated all VM resources (VMs, NIC, Disks) as a single resource and atomically created/deleted them in the driver interface. This caused issues when the controller crashes, during deletions. To fix this, now GetVMs interface checks for all resources instead of just VMs.

### CODE REVIEW

File: pkg/controller/machineset.go [hunk 1]
```diff
@@ -425,11 +425,6 @@ func (c *controller) manageReplicas(allMachines []*v1alpha1.Machine, machineSet
 // meaning it did not expect to see any more of its machines created or deleted. This function is not meant to be
 // invoked concurrently with the same key.
 func (c *controller) reconcileClusterMachineSet(key string) error {
-	startTime := time.Now()
-	defer func() {
```
Comment (hardikdr) [inline pkg/controller/machineset.go]: No harm in having it ryt ?

File: pkg/controller/machine.go [hunk 2]
```diff
@@ -411,7 +413,12 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 				TimeoutActive:  false,
 				LastUpdateTime: metav1.Now(),
 			}
-			machine = c.updateMachineStatus(machine, lastOperation, currentStatus)
+			machine, err = c.updateMachineStatus(machine, lastOperation, currentStatus)
+			if err != nil {
+				// Object no longer exits and has been deleted
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/controller/machine.go]: Done.

File: pkg/controller/machine.go [hunk 3]
```diff
@@ -411,7 +413,12 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
...
+				glog.Warning(err)
+				return nil
+			}
 		}
 
 		var err error
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/controller/machine.go]: Done.

File: pkg/controller/machine.go [hunk 4]
```diff
@@ -393,6 +393,8 @@ func (c *controller) machineUpdate(machine *v1alpha1.Machine, actualProviderID s
 }
 
 func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driver) error {
+	var err error
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/controller/machine.go]: Done.

File: pkg/controller/machineset.go (hunk 1 above)
Comment (prashanth26) [reply pkg/controller/machineset.go]: Restored it with printing at start and end of reconcile.

**Pull request merged**